File: Practice/27/C++/example27/example27/example27.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
using namespace std;
vector <int> last_nums;
void sortdsc() {
	int n = last_nums.size();
	for (int i = 1; i < n; ++i)
	{
		for (int r = 0; r < n - i; r++)
		{
			if (last_nums[r] < last_nums[r + 1])
			{
				// Обмен местами
				int temp = last_nums[r];
				last_nums[r] = last_nums[r + 1];
				last_nums[r + 1] = temp;
			}
		}
	}


}
int main()
{
	setlocale(LC_ALL, "Russian");
//Ввожу кол-во сигналов
	int n;// Кол-во сигналов
	cin >> n;
	if (n>=1 && n <=100000) // соответствие условию
	{
		int go = 1;
		vector<int> nums; //массив n сигналов (значения)
		for (int  i = 0; i < n; i++)
		{

			int temp;
			cin >> temp;
			if (abs(temp) <= pow(10,6))
			{
				nums.push_back(temp); // заполнение массива (значениями)
				
			}
			else {
				cout << "Не соответствует условию";
				go = 0;
			}
			

		}
		if (go == 1)
		{

			for (int i = 0; i < n; i++)// повторит столько сколько сигналов
			{	
				if (last_nums.size() < 5)//До 5 элементов наполнение
				{
					last_nums.push_back(nums[i]);
				}
				if (last_nums.size() == 5)
				{
					for (size_t k = 0; k < 5; k++)
					{
						if (nums[i] <= last_nums[k])
						{
							last_nums[k] = nums[i];
							break;
						}
					}
				}



				sortdsc();
				//тут будет сортировка элементов
				//вывод до 5 сигналов
				for (int j= 0; j < last_nums.size(); j++)
				{
					cout << last_nums[j] << " ";
				}
				cout << endl;

			}
		}

		
		
	}
	else {
		cout << "Не соответствует условию";
	}

}

File: Practice/27/Python/example27.py
def sortdsc(array):
	n = len(array)
	i = 1
	while i < n:
		r = 0
		while r < n-i:
			if int(array[r])<int(array[r+1]):
				temp = array[r]
				array[r] = array[r+1]
				array[r+1] = temp
			r +=1
		i +=1
last_nums = []
n = int(input())
arr = []
if n > 1 and n <= 100000:
	go = 1
	arr = list(input().split())
	for x in arr:
		if abs(int(x)) > 10**6 :
			go = 0
	if go == 1:
		j = 0
		while j<n:
			start = 1
			if len(last_nums) < 5:
				last_nums.append(arr[j])
				start = 0
			if len(last_nums) == 5 and start == 1:
				p = 0
				while p<5:
					if int(arr[j])<=int(last_nums[p]):
						last_nums[p] = arr[j]
						break
					p +=1
			sortdsc(last_nums)
			for x in last_nums:
				print(x,end= " ")
			print()
			j +=1
	else:
		print("Не соответствует условию")
else:
	print("Не соответствует условию")